inc/class-provider.php

```diff
@@ -44,40 +46,60 @@ public function __construct( Factory $factory ) {
 	 *
 	 * @throws Exception If the REST API response could not be decoded.
 	 */
-	protected function request( array $args ): MediaList {
-
-		$args = $this->parse_args( $args );
-
-		$url = get_endpoint();
-		$url = add_query_arg( $args, $url );
-
-		$response = $this->remote_request( $url, [
-			'headers' => [
-				'Accept-Encoding' => 'gzip, deflate',
-				'Connection'      => 'Keep-Alive',
-				'Content-Type'    => 'application/json',
-				'Keep-Alive'      => 30,
-			],
-			'timeout' => 30,
-		] );
-		$response = json_decode( $response );
-
-		if ( json_last_error() ) {
-			throw new Exception( sprintf(
-				/* translators: %s: Error message */
-				__( 'Media error: %s', 'amf-wordpress' ),
-				json_last_error_msg()
-			) );
-		}
-
-		if ( ! is_array( $response ) || ! $response ) {
-			return new MediaList();
-		}
-
-		$items = array_map( [ $this->factory, 'create' ], $response );
-
-		return new MediaList( ...$items );
-	}
+    protected function request( array $args ): MediaList
+    {
+        $args = $this->parse_args( $args );
+
+        // if this is a local multisite, run a query instead of hitting the external API.
+        if ( defined( 'AMF_LOCAL_BLOG_ID' ) {
```

Comment:
The other constant we have is prefixed with `AMF_WORDPRESS_` so it'd good to rename this as `AMF_WORDPRESS_BLOG_ID` or `AMF_WORDPRESS_SITE_ID`